File: Data Structures/Java/String Matching/NaiveSearch.java
import java.util.ArrayList;

public class NaiveSearch{
	
	public static ArrayList<Integer> naiveSearch(String text, String pattern){
		ArrayList<Integer> positions = new ArrayList<Integer>();
		
		int n = text.length();
		int m = pattern.length();
		
		for(int i = 0;i<=n-m;i++){
			int j;
			for(j = 0;j<m;j++){
				if(text.charAt(i+j) != pattern.charAt(j)){
					break;
				}
			}
			if(j == m){
				positions.add(i);
			}
		}
		
		return positions;
	}
	
	public static void main(String[] args){
		String text = "ABCCBACABACABACABACACAACBABAC";
		String pattern = "BAC";
		ArrayList<Integer> positions = naiveSearch(text,pattern);
		System.out.println("Text: "+text);
		System.out.println("Pattern: "+pattern);
		System.out.println();
		
		for(int position: positions){
			System.out.println("Pattern found at position: " + position);
		}
	}
}

File: Algorithms/Sorting Algorithms/C/naive_search.c
#include<stdio.h>
#include<string.h>

int main() {
	
   char mainString[100] = "ABCCBACABACABACABACACAACBABAC";
   char pattern[50] = "BAC";
   int result[strlen(mainString)];
   int index = -1;
   printf("source String: ");
	printf("pattern String: %s \n\n",pattern);
	int patLen = strlen(pattern);
   int strLen = strlen(mainString);
int i;
   for(i = 0; i<=(strLen - patLen); i++) {
      int j;
      for(j = 0; j<patLen; j++) {     
         if(mainString[i+j] != pattern[j])
            break;
      }

      if(j == patLen) { 
         index++;
         result[index] = i;
      }
   }
	
   for(i = 0; i <= index; i++) {
      printf("Pattern found at position: %d\n",result[i]);
   }
}
